packages/curvenote/src/web/deploy.ts

```diff
@@ -211,36 +328,126 @@ export async function promoteContent(session: ISession, cdnKey: string, domains?
   }
 }
 
+type DeploymentStrategy = 'public' | 'private-venue' | 'default-private';
+
+/**
+ * Determine how deployment should be done based on the options and site config
+ *
+ * @returns DeploymentStrategy
+ */
+export function resolveDeploymentStrategy(
+  siteConfig: SiteConfig,
+  opts: { domain?: string; venue?: string },
+): DeploymentStrategy {
+  // if a venue is specified, then it is private and takes precedence over domain
+  if (opts.venue) return 'private-venue';
+
+  const hasDomain = opts.domain ?? siteConfig.domains;
+  if (hasDomain) return 'public';
+
+  // default to private
+  return 'default-private';
+}
+
 export async function deploy(
   session: ISession,
-  opts: Parameters<typeof buildSite>[1] & { ci?: boolean; domain?: string },
+  opts: Parameters<typeof buildSite>[1] & {
+    ci?: boolean;
+    domain?: string;
+    private?: boolean;
+    venue?: string;
+  },
 ): Promise<void> {
   if (session.isAnon) {
     throw new Error(
       '⚠️ You must be authenticated for this call. Use `curvenote token set [token]`',
     );
   }
   const me = await new MyUser(session).get();
-  // Do a bit of prework to ensure that the domains exists in the config file
+  // determine how to deploy based on config and options
   const siteConfig = selectors.selectCurrentSiteConfig(session.store.getState());
   if (!siteConfig) {
     throw new Error('🧐 No site config found.');
   }
+
+  const strategy = resolveDeploymentStrategy(siteConfig, opts);
   const domains = opts.domain ? [opts.domain] : siteConfig?.domains;
-  if (!domains || domains.length === 0) {
-    throw new Error(
-      `🧐 No domains specified, use config.site.domains: - ${me.data.username}.curve.space`,
-    );
+
+  // do confirmation for all strategies up-front
+  // TODO check upload and promotion authorisations up front
+  switch (strategy) {
+    // TODO public-venue?
+    case 'public': {
+      if (!domains || domains.length === 0) {
+        throw new Error(`🚨 Internal Error: No domains specified during public deployment`);
+      }
+      await confirmOrExit(
+        `Deploy local content to "${domains.map((d) => `https://${d}`).join('", "')}"?`,
+        opts,
+      );
+      await preflightPromotePublicContent(session, domains); // TODO check domains exist, and user can promote to them
```

Comment:
Ah, nice, so these preflights are for early reporting in the CLI (i.e. no need to wait for failures). They are not necessary for the actual promotion.

Reply:
yes but they are placeholders at the moment! we need a `check` api call or something to enable these efficiently